src/tree/updater_gpu_hist.cu

```diff
@@ -1009,34 +992,66 @@ class GPUHistMaker : public TreeUpdater {
   void AllReduceHist(int nidx) {
     if (shards_.size() == 1) return;
 
-    reducer_.GroupStart();
-    for (auto& shard : shards_) {
+    monitor_.Start("AllReduce");
+    dh::ExecuteIndexShards(&shards_, [&](int idx, std::unique_ptr<DeviceShard>& shard) {
```

Comment:
I see.  `ExecuteIndexShards` spawns multiple threads so the group calls are no longer needed.